lib/rust/profiler/src/lib.rs

```diff
@@ -479,122 +563,85 @@ where
 ///
 /// # Usage
 ///
-/// The last argument must be an instance of a profiler type. The type given determines the
-/// profiling level at which measurement is enabled; the macro will modify the function's
-/// signature so that it can be called with any profiler that could be a *parent* of the
-/// profiler.
+/// The argument to the macro is a profiler type name, identifying the
+/// [profiling level](#profiling-levels) at which to instrument the function.
 ///
 /// ```
 /// # use enso_profiler as profiler;
 /// # use enso_profiler::profile;
-/// #[profile]
-/// fn small_computation(input: i16, profiler: profiler::Detail) -> u32 {
+/// #[profile(Detail)]
+/// fn example(input: i16) -> u32 {
 ///     todo!()
 /// }
-///
-/// #[enso_profiler_macros::profile]
-/// fn bigger_computation(profiler: profiler::Task) -> u32 {
-///     // Our task-level profiler is not the same type as the detail-level profiler available
-///     // inside `small_computation`; it will be converted implicitly.
-///     small_computation(7, profiler)
-/// }
 /// ```
 ///
 /// This will expand to the equivalent of:
 ///
 /// ```
 /// # use enso_profiler as profiler;
 /// # use enso_profiler::profile;
-/// fn small_computation(input: i16, profiler: impl profiler::Parent<profiler::Detail>) -> u32 {
-///     let _profiler = profiler.new_child("small_computation (file.rs:43)");
-///     let profiler = _profiler.profiler;
-///     return (|input: i16, profiler: profiler::Detail| todo!())(input, profiler);
-/// }
-///
-/// fn bigger_computation(profiler: impl profiler::Parent<profiler::Task>) -> u32 {
-///     let _profiler = profiler.new_child("bigger_computation (file.rs:48)");
-///     let profiler = _profiler.profiler;
-///     return (|profiler: profiler::Task| {
-///         // Our task-level profiler is not the same type as the detail-level profiler available
-///         // inside `small_computation`; it will be converted implicitly.
-///         small_computation(7, profiler)
-///     })(profiler);
+/// fn example(_input: i16) -> u32 {
+///     let __profiler_scope = {
+///         use profiler::Profiler;
+///         let parent = profiler::IMPLICIT_ID;
+///         let now = Some(profiler::Timestamp::now());
+///         let profiler = profiler::Detail::start(parent, "example (profiler/src/lib.rs:78)", now);
+///         profiler::Started(profiler)
+///     };
+///     {
+///         todo!()
```

Comment:
Please does not use `todo!` because it makes harder for us to discover places that needs to be really implemented in the code (searching the codebase for `todo!` should show us all real todos).